### DISCUSSION TITLE
How to apply tailwind classes to dynamic lists?

### DISCUSSION THREAD

**Discussion by Squirrelcoding** [tailwindlabs/tailwindcss#5893] [Help]
In tailwind, there is a feature (which I find ***very*** annoying at times) called preflight that turns most of the HTML elements to appear like normal text. And the documentation says that if you want your lists to remain unchanged, you can just do the following:
```html
<ul className="list-disc list-inside">
  <li>One</li>
  <li>Two</li>
  <li>Three</li>
</ul>
```
But the thing is, I have dynamically generated content and I can't find a way to apply those to it. Does anybody know how this can be done?
I tried wrapping it in a div with the classes but no luck.
```jsx
<div className="list-disc list-inside">
  <ul>
    <li>One</li>
    <li>Two</li>
    <li>Three</li>
  </ul>
</div>
```

**Comment by dance2die** [comment]
Would disabling preflight is an option?   
https://tailwindcss.com/docs/preflight#disabling-preflight

It's for when you add TW to an existing project and when you have existing style (or default browser style).

**Comment by simonswiss** [comment]
Hey @Squirrelcoding!

Two options here:

## 1. Using the Typography plugin

The official [Typography plugin](https://github.com/tailwindlabs/tailwindcss-typography) is specifically designed to help with styling content coming in scenarios where you cannot control the markup output.

Here's a YouTube screencast on how the plugin works:

https://www.youtube.com/watch?v=J0Wy359NJPM

## 2. Disabling Tailwind's Preflight

If you don't want Tailwind's opinionated CSS reset, and want to handle this part yourself, you can opt-out of it in your Tailwind config:

```js
// tailwind.config.js
module.exports = {
  corePlugins: {
    preflight: false
  }
}
```

Hope one of these solutions is elegant enough for your specific use case! ✨

**Comment by MichaelAllenWarner** [comment]
It's not very difficult to make [custom variants](https://tailwindcss.com/docs/plugins#adding-variants) that let you target descendants or children or whatever. For example, if the `ul` will definitely be a direct child of the `div`, then something like this could work:

```js
const plugin = require('tailwindcss/plugin')

module.exports = {
  plugins: [
    plugin(function({ addVariant, e }) {
      addVariant('children', ({ modifySelectors, separator }) => {
        modifySelectors(({ className }) => {
          return `.${e(`children${separator}${className}`)} > *`
        })
      })
    })
  ]
}
```

```html
<div className="children:list-disc children:list-inside">...
```
